mozilla-ai/lumigator
# Batch inference round 1/n

## What's changing

- evaluator has been moved from `mzai/evaluator` to `mzai/jobs/evaluator`
- a new inference job has been added into `mzai/jobs/inference`
- minor fixes: broken evaluator test, minor python version for backend

## How to test it

- backend, sdk: `uv run pytest` (strangely SDK still appears to be running ok with the mock `/experiments` endpoint?)
- evaluator: `uv venv --python 3.11.9 && uv pip install pytest && uv pip install -r requirements.txt` to set the environment, then `uv run pytest`
- inference: no tests yet

## Additional notes for reviewers

This is the first of a few  PRs to split the larger task into smaller ones:
- move evaluator, wire an inference job into lumigator [this PR]
- update the inference job configuration (as right now it takes everything that is provided to evaluation and just ignores unuseful params e.g. metrics) and the respective API
- improve the inference job code (using vLLM-as-a-library and relying on ray.data for further parallelization)
- fix outputs so an updated CSV file is returned instead of a JSON (this requires a discussion to choose whether/where we want to store further metadata)
- unit tests

## I already...
